core/src/main/scala/cr/pulsar/Producer.scala

```diff
@@ -127,21 +127,15 @@ object Producer {
             }
           }
 
-          override def send(msg: E): F[MessageId] = {
-            val event = serialise(msg)
+          override def send(msg: E): F[MessageId] =
+            logAction(msg)(topic.url) &> blocker.delay {
+                  buildMessage(serialise(msg), shardKey(msg)).send()
```

Comment:
Nitpicking: formatting here and below

Reply:
That's how the formatter did it :man_shrugging: